### PR TITLE
Add a default module manifest #24

### PR DESCRIPTION
**Pull request by KevinMarquette**
Add a minimally viable module manifest to the PSModule template. 
This allows end users to create a new module from the template, build it, then import the psd1.

### CODE REVIEW

**Comment by tylerl0706** [comment]
This is AWESOME. Github doesn't seem to want to render it in the PR... But clicking on the file showed it. I have one comment... 

On the `RootModule` field, you set it to bin/netstandard2.0/etc... That's totally fine but I think that might give the wrong idea to folks on how to ship their modules. 

If the module manifest is sitting at the root of their repo, the user might accidentally ship their code in addition to their build output.

What I would recommend is to set the RootModule to `Microsoft.PowerShell.Standard.Module.Template.dll` and then in the csproj, copy the psd1 to the build/publish directory.

Heres an example:


```
<None Include="foo.psd1">
  <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
</None>
```

**Comment by KevinMarquette** [comment]
Good idea. I overlooked that option.

**Comment by KevinMarquette** [comment]
The suggested change has been made and tested. I also changed the file encoding on the psm1 file to be UTF-8 no BOM.

File: src/dotnetTemplate/Microsoft.PowerShell.Standard.Module.Template/Microsoft.PowerShell.Standard.Module.Template/Microsoft.PowerShell.Standard.Module.Template.psd1 [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+# NestedModules = @()
+
+# Functions to export from this module, for best performance, do not use wildcards and do not delete the entry, use an empty array if there are no functions to export.
+FunctionsToExport = @()
+
+# Cmdlets to export from this module, for best performance, do not use wildcards and do not delete the entry, use an empty array if there are no cmdlets to export.
+CmdletsToExport = @('Test-SampleCmdlet')
+
+# Variables to export from this module
+VariablesToExport = '*'
```
Comment (bergmeister) [inline src/dotnetTemplate/Microsoft.PowerShell.Standard.Module.Template/Microsoft.PowerShell.Standard.Module.Template/Microsoft.PowerShell.Standard.Module.Template.psd1]: PSScriptAnalyzer will not like that: [UseToExportFieldsInManifest](https://github.com/PowerShell/PSScriptAnalyzer/blob/development/RuleDocumentation/UseToExportFieldsInManifest.md)

Comment (bergmeister) [inline src/dotnetTemplate/Microsoft.PowerShell.Standard.Module.Template/Microsoft.PowerShell.Standard.Module.Template/Microsoft.PowerShell.Standard.Module.Template.psd1]: PSScriptAnalyzer will not like that: [UseToExportFieldsInManifest](https://github.com/PowerShell/PSScriptAnalyzer/blob/development/RuleDocumentation/UseToExportFieldsInManifest.md)

File: src/dotnetTemplate/Microsoft.PowerShell.Standard.Module.Template/Microsoft.PowerShell.Standard.Module.Template/Microsoft.PowerShell.Standard.Module.Template.csproj [hunk 2]
```diff
@@ -9,6 +9,12 @@
...
       <PrivateAssets>All</PrivateAssets>
     </PackageReference>
+
+    <None Include="Microsoft.PowerShell.Standard.Module.Template.psd1">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
   </ItemGroup>
 
+  
+
```
[the remark this answers is not in the document]
Comment (KevinMarquette) [reply src/dotnetTemplate/Microsoft.PowerShell.Standard.Module.Template/Microsoft.PowerShell.Standard.Module.Template/Microsoft.PowerShell.Standard.Module.Template.csproj]: corrected

File: src/dotnetTemplate/Microsoft.PowerShell.Standard.Module.Template/Microsoft.PowerShell.Standard.Module.Template/Microsoft.PowerShell.Standard.Module.Template.psd1 (hunk 1 above)
Comment (KevinMarquette) [reply src/dotnetTemplate/Microsoft.PowerShell.Standard.Module.Template/Microsoft.PowerShell.Standard.Module.Template/Microsoft.PowerShell.Standard.Module.Template.psd1]: corrected

**Comment by KevinMarquette** [comment]
I cleaned up the extra whitespace and made corrections for script analyzer. I ran it on the resulting module without any warnings.

I also confirmed the manifest was renamed.

    PS> Get-ChildItem

    Mode                LastWriteTime         Length Name
    ----                -------------         ------ ----
    -a----         8/7/2018   9:38 AM           2030 testmodule.deps.json
    -a----         8/7/2018   9:38 AM           5120 testmodule.dll
    -a----         8/7/2018   9:38 AM            860 testmodule.pdb
    -a----         8/7/2018   9:37 AM           3849 testmodule.psd1
    

    PS>Invoke-ScriptAnalyzer .\testmodule.psd1
    PS>Import-Module .\testmodule.psd1
    PS>Get-Module testmodule

    ModuleType Version    Name                                ExportedCommands
    ---------- -------    ----                                ----------------
    Binary     0.1.0      testmodule                          Test-SampleCmdlet

**Pull request merged**